### ISSUE TITLE
404 errors in loading chunk.js files

### ISSUE THREAD

**Issue by ally-ssa** [ADB-SPADE/geonode-spade#1]
## Expected Behavior
After starting the Django server following the steps of starting GeoNode locally in development mode, when I open http://localhost:8001/ I should be able to see a blank GeoNode page with a "Welcome to GeoNode" message, as in
![image](https://user-images.githubusercontent.com/100031007/236108078-7ae9d2c6-e472-4b59-8a86-2e7743ea70fc.png)

## Actual Behavior
http://localhost:8001/ shows a blank container under the navbar (see example below):
![image](https://user-images.githubusercontent.com/100031007/236109044-d73badfb-3f63-43f7-a4e1-af2804a7916f.png)

 in addition, I see the following in DevTools:
![image](https://user-images.githubusercontent.com/100031007/236109068-426ec1da-c34e-4981-97af-8e4eb0b8dd9d.png)

When I go to another page, for example Maps, the page is stuck on loading screen:
![image](https://user-images.githubusercontent.com/100031007/236112648-7279efd1-e58e-464d-a1b8-0924ea28b41e.png)

And I also see 404 errors in DevTools:
![image](https://user-images.githubusercontent.com/100031007/236112733-70dc228e-2396-4125-b82c-ffdb8513a9f5.png)

## Steps to Reproduce the Problem

  1. I installed `Docker` and `docker-compose` using [GeoNode documentation](https://docs.geonode.org/en/4.x/install/advanced/core/index.html#docker-setup-first-time-only) as reference
  2. I cloned the `geonode-spade` repo then checked out the `spade-for-isms` branch
  3. I loaded the submodules; I was also having problems loading the `safeguard-screening-map` submodule, though I was able to load it using
```.sh
git submodule update --force --recursive --init --remote
````
  4. I ran the following commands, following _Start locally in development mode_ of [README_EGIS.md](https://github.com/ADB-SPADE/geonode-spade/blob/spade-for-isms/README_EGIS.md#start-locally-in-development-mode)
```.sh
docker-compose --env-file .env_dev_egis -f docker-compose.yml -f ./docker-compose.egis-local.dev.yml build

# Docker UP
docker-compose --env-file .env_dev_egis -f docker-compose.yml -f ./docker-compose.egis-local.dev.yml up -d

# Enter django container shell
docker exec -it django4geonode /bin/bash

# Start Django server in development mode
python manage.py runserver 0.0.0.0:8001
```
  5. I opened http://localhost:8001/ in my browser (tested in Edge and Chrome)

## Specifications
I am on Windows 10, but I run `Docker` and `docker-compose` in WSL 2 with the Ubuntu 22.04 LTS distro. Here are other details about my system:
- Docker Engine version: 23.0.5
- Docker Compose version: 2.17.3

**Comment by ewsterrenburg** [comment]
@ally-ssa Please let me know if the following works for you:

# Prerequisites:

- Make sure you have a Github Personal Access Token (https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token). 
- Make sure the docker images, volumes, build cache, etc. is all cleaned

When cloning the private modules, this token functions as the password.

# Fetching the source code

- Clone the repository and its submodules: `git clone --branch spade-for-isms --recurse-submodules https://github.com/ADB-SPADE/geonode-spade.git`
- Login with github username and the token mentioned above
- Open the directory: `cd geonode-spade`
- Open VS Code e `code .`

# Creating a devcontainer

Now VS Code should give the option to "Reopen in Container".  Now a docker-compose project will be set up, based on the contents of the .devcontainer folder.

![image](https://user-images.githubusercontent.com/1898044/236908898-2ee56ee5-b692-47bd-b593-1ed93903b40e.png)

# Running the application

In the django4geonode container (VS Code terminal should open here automatically), run `python manage.py runserver 0.0.0.0:8000`

Now the whole application should be available on http://localhost with the integration between Django & GeoServer intact.
The Django application should also be available on http://localhost:8000.

# Taming the memory
- depending on the parts being developed, gsconf4geonode, rabbitmq4geonode, geoserver4geonode & celery4geonode can be paused to keep memory usage a bit down.

**Comment by ally-ssa** [comment]
Hi @ewsterrenburg, thanks for the comment. I was able to clone the source code including the submodules without problems this time, but I encounter this error when reopening `geonode-spade` in a dev container:
![image](https://user-images.githubusercontent.com/100031007/236991075-1bcafc3b-0640-4ce2-b0f6-e4fa38a60019.png)
There seems to be a permission error, even though the directory where `geonode-spade` is located has permission 775

**Comment by ewsterrenburg** [comment]
@ally-ssa  https://stackoverflow.com/questions/38882654/docker-entrypoint-running-bash-script-gets-permission-denied, https://stackoverflow.com/questions/72883508/unable-to-start-container-process-permission-denied-unknown, would it help to make the entrypoint shellscript executable?

**Comment by ally-ssa** [comment]
Thanks, got the permission issue resolved by running 
```.sh
sudo chmod 775 entrypoint.sh
```
Unfortunately the 404 issue is still there ☹
![image](https://github.com/ADB-SPADE/geonode-spade/assets/100031007/06914c2d-415b-42ce-af06-809c86094dd2)

**Comment by ally-ssa** [comment]
Discovered that the app was encountering issues when run in WSL on port 8000, and using one of these workarounds helped:

1. In the dev container terminal, I ran the app using

```.sh
python manage.py runserver 0.0.0.0:9000
```
which allowed me to access the app using http://localhost:9000. Note from @ewsterrenburg that dev container .env and nginx Dockerfile may have to be updated to change the references to port 8000 to 9000

2. In the dev container terminal, I ran the app using

```.sh
python manage.py runserver 0.0.0.0:8000
```
but I access the app in browser using `http://<WSL_IP_ADDRESS>:8000` instead of http://localhost:8000. The WSL IP address can be determined using `ip addr show eth0` or `hostname -I`
